jwala-integration-test/src/test/postman/jwala-collection.postman_collection.json

```diff
@@ -2733,15 +2733,12 @@
 								"exec": [
 									"tests[\"Status code is 200\"] = responseCode.code === 200;",
 									"",
-									"function pausecomp(millis)",
+									"function mockSleep(millis)",
 									" {",
-									"  var date = new Date();",
-									"  var curDate = null;",
-									"  do { curDate = new Date(); }",
-									"  while(curDate-date < millis);",
+									"return new Promise(resolve => setTimeout(resolve, millis))",
```

Comment:
Promises and setTimeout are non-blocking - they execute asynchronously but they don't pause the thread. This solution won't have the intended delay that you're looking for.

See the answer here for more explanation:
https://stackoverflow.com/questions/36588775/are-javascript-promise-asynchronous